test/publish.sh

```diff
@@ -27,7 +27,39 @@ else
   echo "No credential helpers found, push to docker may not function properly"
 fi
 
-DOCKER_HUB=${DOCKER_HUB:-gcr.io/istio-testing}
+function cleanup() {
+  # shellcheck disable=SC2046
+  docker stop $(docker ps -a -q --filter label=istio-release-builder)
+}
+trap cleanup EXIT
+
+# Setup fake GCS and registry
+docker run -d  --rm  \
+  -p "7480:5000" --label istio-release-builder \
+  --name "release-builder-registry" \
+  gcr.io/istio-testing/registry:2
+docker run -d  --rm  \
+  -p "7481:7481" --label istio-release-builder \
+  --name "release-builder-gcs" \
+  gcr.io/istio-testing/fake-gcs-server:1.45.2 \
+  -scheme http -port 7481
+
+# Setup our bucket. Add retry since the registry may not be ready yet
+counter=0
+while : ; do
+  [[ "$counter" == 10 ]] && exit 1
+  curl -X POST \
+    -d '{"name":"istio-build"}' \
+    -H "content-type: application/json" \
+    -H "accept: application/json" \
+    'http://127.0.0.1:7481/storage/v1/b?alt=json&project=test&projection=full' && break
+   sleep 1
+   echo "Trying again... Try #$counter"
+   counter=$((counter+1))
+done
+
+DOCKER_HUB=${DOCKER_HUB:-"localhost:7480"}
+export GCS_HOST=${GCS_HOST-"http://localhost:7481"}
```

Comment:
```suggestion
export GCS_HOST=${GCS_HOST:-"http://localhost:7481"}
```

Reply:
This was intended to allow users to write `GCS_HOST=""` to not used it. But I guess I later check if `host != ""` not if its not found...

Reply:
Oh actually this is still needed, your suggestion will overwrite `GCS_HOST=""` to `GCS_HOST="localhost:7481"`. The current one will not, but if GCS_HOST is unset it will set it.

TBH I am not sure we need any configuration here since its the test one...